core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStandaloneTrainScheduleParser.java

```diff
@@ -69,52 +70,42 @@ private static Allowance parseAllowance(
             RJSAllowance rjsAllowance
     ) throws InvalidSchedule {
 
-        if (rjsAllowance.getClass() == RJSAllowance.Construction.class) {
-            var rjsConstruction = (RJSAllowance.Construction) rjsAllowance;
-            if (Double.isNaN(rjsConstruction.beginPosition))
-                throw new InvalidSchedule("missing construction allowance begin_position");
-            if (Double.isNaN(rjsConstruction.endPosition))
-                throw new InvalidSchedule("missing construction allowance end_position");
-            return new MarecoAllowance(
-                    new EnvelopeSimContext(rollingStock, envelopePath, timeStep),
-                    rjsConstruction.beginPosition,
-                    Math.min(envelopePath.length, rjsConstruction.endPosition),
-                    getPositiveDoubleOrDefault(rjsConstruction.capacitySpeedLimit, 30 / 3.6),
-                    List.of(
-                            new AllowanceRange(
-                                    rjsConstruction.beginPosition,
-                                    rjsConstruction.endPosition,
-                                    parseAllowanceValue(rjsConstruction.value)
-                            )
-                    )
-            );
+        var allowanceDistribution = rjsAllowance.distribution;
+        double beginPos;
+        double endPos;
+        List<AllowanceRange> ranges;
+        // parse allowance type
+        if (rjsAllowance instanceof RJSAllowance.EngineeringAllowance engineeringAllowance) {
+            beginPos = engineeringAllowance.beginPosition;
```

Comment:
Apparently these lines are not covered by tests, I'll right one to make code coverage work